### PR TITLE
form fixes

### PR DESCRIPTION
**Pull request by EricWVGG**
- huge form fixes (including "text capture" infinite re-render bug)
- better handling for setting initialState
- removed a couple deprecated components

### CODE REVIEW

File: src/components/Form/index.tsx [hunk 1]
```diff
@@ -42,15 +44,15 @@ const Form = <TStateInterface extends Record<string, unknown>>({
   ...props
 }: FormProps<TStateInterface>) => {
   const [disabled, setDisabled] = useState(true)
-  const [formState, setFormState] = useState<Pick<JsonFormsCore, 'data' | 'errors'>>()
```
Comment (EricWVGG) [inline src/components/Form/index.tsx:45]: removing this is the real fix.

File: src/components/CreateGrants/NewGrantForm/index.tsx [hunk 2]
```diff
@@ -28,39 +29,40 @@ const NewGrantForm = () => {
...
-    //     date: date.format('YYYY-MM-DD HH:mm:ss'),
-    //   },
-    // })
+    setData({
+      ...initialState,
+      grant: {
+        ...initialState.grant,
+        date: date.format('YYYY-MM-DD HH:mm:ss'),
+      },
+    })
```
Comment (EricWVGG) [inline src/components/CreateGrants/NewGrantForm/index.tsx:43]: restores the pre-fill for the Grant Date input (which, yes, is read-only)

File: src/components/CreateGrants/NewGrantForm/index.tsx [hunk 3]
```diff
@@ -28,39 +29,40 @@ const NewGrantForm = () => {
...
+      errors.push({
+        instancePath: '/project/walletAddress',
+        message: 'Invalid blockchain address',
+        schemaPath: '#/properties/project/properties/walletAddress',
+        keyword: '',
+        params: {},
+      })
+    }
+    setAdditionalErrors(errors)
+  }, [debouncedData])
```
Comment (EricWVGG) [inline src/components/CreateGrants/NewGrantForm/index.tsx:62]: restores blockchain wallet address validation. (only for the grant form; I haven't found a way to perform this validation on the user insertion array yet, but as I mentioned during the call we really shouldn't be inserting users there in the first place)

File: src/components/CreateGrants/NewGrantForm/index.tsx [hunk 4]
```diff
@@ -177,9 +179,13 @@ const FormWrapper = styled.div`
...
-    td:nth-child(7),
-    th:nth-child(7) {
-      display: none;
+    td:nth-child(7) {
+      padding: 5px;
+      background-color: rgba(255, 255, 255, 0.1);
+      color: white;
+      font-weight: 300;
+      font-size: 9px;
+      pointer-events: none;
```
Comment (EricWVGG) [inline src/components/CreateGrants/NewGrantForm/index.tsx:188]: restores error messaging for insert-users array of objects

File: src/components/Form/Controls/StringControl/index.tsx [hunk 5]
```diff
@@ -39,13 +39,36 @@ export const StringControl = ({
...
+   * That error goes away when we go unmanaged, but then we're not able to do special formatting
+   *   like toLowerCase()
+   * So, this is the compromise. Not crazy about it. :\ -EJ
+   */
+  const [localData, setLocalData] = useState<string>(data || '')
+  useEffect(() => {
+    if (!localData) return
+    handleChange(path, localData)
+  }, [localData])
+
```
Comment (EricWVGG) [inline src/components/Form/Controls/StringControl/index.tsx:53]: fixes unmanaged-input error

File: src/components/SettingsParts/EditProfile/index.tsx [hunk 6]
```diff
@@ -14,9 +14,12 @@ const EditProfile = () => {
   const [updateUser] = useMutation(UPDATE_USER)
   const loggedInUser = useReactiveVar(loggedInUserVar)
 
-  const [data, setData] = useState<FormState>(initialState)
+  const [data, setData] = useState<FormState>()
   useEffect(() => {
+    // set initialState
+    if (!!data || !loggedInUser) return
     setData({
+      ...initialState,
```
Comment (EricWVGG) [inline src/components/SettingsParts/EditProfile/index.tsx:22]: better feeding of `initialState` to the form data

File: src/components/SettingsParts/EditSettings/index.tsx [hunk 7]
```diff
@@ -11,9 +11,12 @@ const EditSettings = () => {
   const [updateUser] = useMutation(UPDATE_USER)
   const loggedInUser = useReactiveVar(loggedInUserVar)
 
-  const [data, setData] = useState<FormState>(initialState)
+  const [data, setData] = useState<FormState>()
   useEffect(() => {
+    // set initialState
+    if (!!data || !loggedInUser) return
     setData({
+      ...initialState,
```
Comment (EricWVGG) [inline src/components/SettingsParts/EditSettings/index.tsx:19]: better feeding of `initialState` to the form data

File: src/components/CreateGrants/NewGrantForm/index.tsx (hunk 2 above)
Comment (geometrizen) [reply src/components/CreateGrants/NewGrantForm/index.tsx:43]: I approve this change! 👍

File: src/components/Form/Controls/StringControl/index.tsx (hunk 5 above)
Comment (geometrizen) [reply src/components/Form/Controls/StringControl/index.tsx:53]: thank you for calling this out-- does this affect whether we are doing `toLowerCase()` for a wallet address? (wallet addresses are sometimes lowercase and sometimes mixed case and our convention is to convert to lower case to store in the db)

File: src/components/Form/Controls/StringControl/index.tsx [hunk 8]
```diff
@@ -39,13 +39,36 @@ export const StringControl = ({
...
   const [parsedErrors, setParsedErrors] = useState<string[]>([])
   useEffect(() => {
     const splitErrors = errors?.split('\n').filter(e => e !== '') || []
     if (required && (!data || data === '')) splitErrors.push('is a required property')
     setParsedErrors(splitErrors)
   }, [errors, required, data])
 
+  const handleStringChange = (value: string) => {
+    // put any special formatting rules here
+    if (uischema?.options?.format === 'creditCard') {
```
Comment (geometrizen) [inline src/components/Form/Controls/StringControl/index.tsx:63]: just wondering why there's a credit card thing here? 🤔

File: src/components/Form/Controls/StringControl/index.tsx [hunk 9]
```diff
@@ -103,31 +113,30 @@ export const StringControl = ({
...
           <input
             {...{ required, autoComplete }}
             disabled={!enabled || processing}
             minLength={schema?.minLength}
             maxLength={schema?.maxLength}
-            type={uischema?.options?.format || 'text'}
+            type="date"
             placeholder={uischema?.options?.placeholder || ' '}
-            defaultValue={data}
-            onChange={e => handleChange(path, ccnFormat(e.target.value))}
```
Comment (geometrizen) [inline src/components/Form/Controls/StringControl/index.tsx:128]: seems like maybe the creditcard/ccn stuff was a just placeholder? (see above comment)

File: src/components/Form/index.tsx (hunk 1 above)
Comment (geometrizen) [reply src/components/Form/index.tsx:45]: I am a fan of the real fix 👍

File: src/forms/createGrants.ts [hunk 10]
```diff
@@ -265,7 +262,7 @@ export const uischema = {
               type: 'Control',
               scope: '#/properties/grant/properties/date',
               options: {
-                // readonly: true,
+                readonly: true,
```
Comment (geometrizen) [inline src/forms/createGrants.ts:265]: 👍

File: src/forms/createGrants.ts [hunk 11]
```diff
@@ -351,6 +348,9 @@ export const uischema = {
                 {
                   type: 'Control',
                   scope: '#/properties/project/properties/walletAddress',
+                  options: {
+                    format: 'lowercase',
+                  },
```
Comment (geometrizen) [inline src/forms/createGrants.ts:353]: oic okay cool 👍 I'm guessing this means that we convert wallet address to lower case, yeah?

**Review by geometrizen — CHANGES_REQUESTED** [review]
looks good overall but there are maybe a couple things to clean up before we merge-- please see inline comments. also wondering about all of the deleted files but i'm assuming that's part of the general somewhat large refactor you mentioned earlier today 👍

**Pull request merged**